src/System.CommandLine.Hosting.Tests/HostingHandlerTest.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System.CommandLine.Binding;
+using System.CommandLine.Builder;
+using System.CommandLine.Invocation;
+using System.CommandLine.IO;
+using System.CommandLine.Parsing;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+
+namespace System.CommandLine.Hosting.Tests
+{
+    public static class HostingHandlerTest
+    {
+
+        [Fact]
+        public static void Constructor_Injection_Injects_Service()
+        {
+            var service = new MyService();
+
+            var parser = new CommandLineBuilder(
+                new MyCommand()
+                )
+                .UseHost((builder) => {
+                    builder.ConfigureServices(services =>
+                    {
+                        services.AddTransient(x => service);
+                    })
+                    .UseCommandHandler<MyCommand, MyCommand.MyHandler>();
+                })
+                .Build();
+
+            var result = parser.InvokeAsync(new string[] { "--int-option", "54"})
```

Comment:
You can change the method to `async Task` and then `await` this, which is a bit easier to read I think.